pkg/types/v2/structure.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleCloudPlatform/container-structure-test/pkg/drivers"
+	types "github.com/GoogleCloudPlatform/container-structure-test/pkg/types/unversioned"
+)
+
+type StructureTest struct {
+	DriverImpl         func(drivers.DriverConfig) (drivers.Driver, error)
+	DriverArgs         drivers.DriverConfig
+	GlobalEnvVars      []types.EnvVar      `yaml:"globalEnvVars"`
+	CommandTests       []CommandTest       `yaml:"commandTests"`
+	FileExistenceTests []FileExistenceTest `yaml:"fileExistenceTests"`
+	FileContentTests   []FileContentTest   `yaml:"fileContentTests"`
+	MetadataTest       MetadataTest        `yaml:"metadataTest"`
+	LicenseTests       []LicenseTest       `yaml:"licenseTests"`
+}
+
+func (st *StructureTest) NewDriver() (drivers.Driver, error) {
+	return st.DriverImpl(st.DriverArgs)
+}
+
+func (st *StructureTest) SetDriverImpl(f func(drivers.DriverConfig) (drivers.Driver, error), args drivers.DriverConfig) {
+	st.DriverImpl = f
+	st.DriverArgs = args
+}
+
+func (st *StructureTest) RunAll() []*types.TestResult {
+	results := make([]*types.TestResult, 0)
+	results = append(results, st.RunCommandTests()...)
+	results = append(results, st.RunFileExistenceTests()...)
+	results = append(results, st.RunFileContentTests()...)
+	results = append(results, st.RunLicenseTests()...)
+	return results
+}
+
+func (st *StructureTest) RunCommandTests() []*types.TestResult {
```

Comment:
do structure tests implement an interface? this function looks similar to the other ones, can you share some of this implementation with the other kinds of tests? 

1. Validate
2. New Driver
3. defer destroy
4. Setup
5. defer Teardown

Reply:
yeah, this needs a rework. I opened an issue a few days ago about having a way to persist the driver between test runs, so when I get into that I'm gonna need to refactor the test driver. I'll leave this for now, and overhaul it in a separate PR.